packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/summary.js

```diff
@@ -17,72 +18,104 @@ const Summary = props => {
 
     const talonProps = useSummary({ ...summaryOperations });
 
-    const { billingAddress, isBillingAddressSame, paymentNonce } = talonProps;
+    const {
+        billingAddress,
+        isBillingAddressSame,
+        isLoading,
+        paymentNonce,
+        selectedPaymentMethod
+    } = talonProps;
 
-    const paymentSummary = `${paymentNonce.details.cardType} ending in ${
-        paymentNonce.details.lastFour
-    }`;
+    if (isLoading) {
+        return (
+            <LoadingIndicator classes={{ root: classes.loading }}>
+                Fetching Payment Information
+            </LoadingIndicator>
+        );
+    }
 
-    const billingAddressSummary =
-        !isBillingAddressSame && billingAddress ? (
-            <div className={classes.address_summary_container}>
-                <div>
-                    <span className={classes.first_name}>
-                        {billingAddress.firstName}
-                    </span>
-                    <span className={classes.last_name}>
-                        {billingAddress.lastName}
-                    </span>
+    if (selectedPaymentMethod.code !== 'braintree') {
```

Comment:
I think we should rethink payment info selection and summary, but maybe not in this PR. Consider what happens if you have multiple options like braintree or check. We hardcoded the summary and edit forms to display only braintree, which means there is no way to change payment method once you've selected one.